widget/fileicon.go

```diff
@@ -48,23 +48,9 @@ func (i *FileIcon) setURI(uri fyne.URI) {
 		return
 	}
 
-	switch splitMimeType(uri) {
-	case "application":
-		i.resource = theme.FileApplicationIcon()
-	case "audio":
-		i.resource = theme.FileAudioIcon()
-	case "image":
-		i.resource = theme.FileImageIcon()
-	case "text":
-		i.resource = theme.FileTextIcon()
-	case "video":
-		i.resource = theme.FileVideoIcon()
-	default:
-		i.resource = theme.FileIcon()
-	}
-
 	i.URI = uri
 	i.cachedURI = nil
+	i.resource = i.setIcon(i.URI)
```

Comment:
This is not a setter, how about `lookupIcon`?

Reply:
Right, good point. Will change that. It was a setter initially :)